lib/cocoapods-core/specification/dsl.rb

```diff
@@ -76,8 +76,10 @@ module DSL
       #   @param  [String] name
       #           the name of the pod.
       #
-      root_attribute :name,
-                     :required => true
+      attribute :name,
```

Comment:
That's not required to lint `Specification`, because the names are not flattened. Subspecs store only the last name element in their `attributes_hash[:name]`.